apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/Operations.java

```diff
@@ -201,22 +200,18 @@ public void deleteStagedOrphanDirectory(
     }
   }
 
-  /** Run ExpireSnapshots operation on a given fully-qualified table name. */
-  public ExpireSnapshots.Result expireSnapshots(String fqtn, long expireBeforeTimestampMs) {
-    return expireSnapshots(getTable(fqtn), expireBeforeTimestampMs);
+  /** Expire snapshots on a given fully-qualified table name. */
+  public void expireSnapshots(String fqtn, long expireBeforeTimestampMs) {
+    expireSnapshots(getTable(fqtn), expireBeforeTimestampMs);
   }
 
-  /** Run ExpireSnapshots operation on a given {@link Table}. */
-  public ExpireSnapshots.Result expireSnapshots(Table table, long expireBeforeTimestampMs) {
-    return SparkActions.get(spark)
-        .expireSnapshots(table)
+  /** Expire snapshots on a given {@link Table}. */
+  public void expireSnapshots(Table table, long expireBeforeTimestampMs) {
+    table
+        .expireSnapshots()
+        .cleanExpiredFiles(false)
         .expireOlderThan(expireBeforeTimestampMs)
-        .deleteWith(
-            (file) -> {
-              // skip deletion
-              log.info("Detected file {} that is not part of survived snapshots", file);
-            })
-        .execute();
+        .commit();
```

Comment:
good idea to use commit instead of execute, this avoids needing to override the default behavior of `BaseExpireSnapshotsSparkAction`